incydr/cli/cmds/alerts.py

```diff
@@ -0,0 +1,265 @@
+    for name, _field in alert_dict.items():
+        if name == "id":
+            continue
+        if name == "note.message" and alert.note.message is not None:
+            field_table.add_row(
+                Panel(
+                    Markdown(alert.note.message, justify="left"),
+                    title="Notes",
+                    width=80,
+                )
+            )
+        else:
+            field_table.add_row(f"{name} = {rgetattr(alert, name)}")
+    console.print(Panel.fit(field_table))
+
+
+@click.group(cls=IncydrGroup)
+def alerts():
+    pass
+
+
+@alerts.command(cls=IncydrCommand)
+@click.pass_context
+@columns_option
+@table_format_option
+@output_options
+@advanced_query_option
+@filter_options
+def search(
+    ctx: Context,
+    format_: TableFormat,
+    columns: Optional[str],
+    output: Optional[str],
+    certs: Optional[str],
+    ignore_cert_validation: Optional[bool],
+    advanced_query: Optional[str],
+    start: Optional[str],
+    end: Optional[str],
+    on: Optional[str],
+    alert_id: Optional[str],
+    type_: Optional[str],
+    name: Optional[str],
+    actor: Optional[str],
+    actor_id: Optional[str],
+    risk_severity: Optional[str],
+    state: Optional[str],
+    rule_id: Optional[str],
+    alert_severity: Optional[str],
+):
+    """
+    Search alerts.
+    """
+    if advanced_query:
+        query = AlertQuery.parse_raw(advanced_query)
+    else:
+        if not any([start, on]):
+            raise BadOptionUsage(
+                "start",
+                "--start or --on options are required if not using the --advanced-query option.",
+            )
+        query = _create_query(
+            start=start,
+            end=end,
+            on=on,
+            alert_id=alert_id,
+            type_=type_,
+            name=name,
+            actor=actor,
+            actor_id=actor_id,
+            risk_severity=risk_severity,
+            state=state,
+            rule_id=rule_id,
+            alert_severity=alert_severity,
+        )
+
+    client = ctx.obj()
+    if not query.tenant_id:
+        query.tenant_id = client.tenant_id
+
+    # TODO: should we use iter_all here?
+    alerts_ = client.session.post("/v1/alerts/query-alerts", json=query.dict())
+    alerts_ = alerts_.json()["alerts"]
+
+    # TODO: with outputting refactor
+    # this is definitely not a permanent fix.  Not all alert objects contain all the keys possible.
+    # This converts the first alert to a dict with all fields so that CSV output knows about all of them
+    alerts_ = [json.loads(AlertSummary.parse_obj(alerts_[0]).json())] + alerts_[1:]
+
+    if output:
+        output_format_logger(alerts_, output, columns, certs, ignore_cert_validation)
+    else:
+        output_response_format(
+            alerts_,
+            "Alerts",
+            format_,
+            columns,
+            client.settings.use_rich,
+        )
+
+
+# TODO - single or multiple alerts for show details?
```

Comment:
^

Reply:
Let's leave it single for now, although I do like the idea of being able to do `incydr alerts search | incydr alerts show -` to be able to get a nice human-readable summary of a large number of alerts.